src/viriatum_commons/test/report.c

```
@@ -0,0 +1,365 @@
+
+ERROR_CODE write_junit_report(struct test_report_t *report, FILE *file) {
+    size_t index;
+    struct test_result_t *result;
+    char escaped[TEST_ESCAPE_SIZE];
+
+    fprintf(file, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n");
+    fprintf(file, "<testsuites>\n");
+
+    escape_xml_report(report->name, escaped, TEST_ESCAPE_SIZE);
+    fprintf(
+        file,
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)